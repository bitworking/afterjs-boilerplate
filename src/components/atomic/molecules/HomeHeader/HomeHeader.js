// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import logo from '../../../../assets/react.svg';

type Props = {
  /** please describe the prop children */
  children?: Node,
}

const HomeHeaderStyled = styled.div`
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
  
  img {
    animation: logo-spin infinite 20s linear;
    height: 80px;
  }
  
  @keyframes logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`;

const HomeHeader = (props: Props): Node => (
  <HomeHeaderStyled>
    <img src={logo} alt="logo" />
    <h2>Welcome to After.js</h2>
  </HomeHeaderStyled>
);

HomeHeader.defaultProps = {

}; 

export default HomeHeader;
