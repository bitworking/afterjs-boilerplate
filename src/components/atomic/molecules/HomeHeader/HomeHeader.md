## Demo of the molecule HomeHeader

    <HomeHeader

    >
       Lorem Ipsum Children
    </HomeHeader>
    
Please add more examples    
