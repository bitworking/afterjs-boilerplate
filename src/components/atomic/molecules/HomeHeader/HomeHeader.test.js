import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import HomeHeader from './HomeHeader';

it('renders default HomeHeader snapshot', () => {
  const tree = renderer.create(
    <HomeHeader

    >
       Lorem Ipsum Children
    </HomeHeader>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
