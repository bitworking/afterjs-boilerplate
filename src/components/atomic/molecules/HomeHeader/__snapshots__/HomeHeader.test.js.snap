// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default HomeHeader snapshot 1`] = `
.c0 {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.c0 img {
  -webkit-animation: logo-spin infinite 20s linear;
  animation: logo-spin infinite 20s linear;
  height: 80px;
}

<div
  className="c0"
>
  <img
    alt="logo"
    src="react.svg"
  />
  <h2>
    Welcome to After.js
  </h2>
</div>
`;
